examples/helloworld/helloworldagent.go

```diff
@@ -0,0 +1,154 @@
+	"k8s.io/client-go/informers"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1lister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+// Helloworld Agent is an example that syncs configmap in cluster namespace of hub cluster
+// to the install namespace in managedcluster.
+
+func newAgentCommand() *cobra.Command {
+	o := NewAgentOptions()
+	cmd := controllercmd.
+		NewControllerCommandConfig("addon-agent", version.Get(), o.RunAgent).
+		NewCommand()
+	cmd.Use = "agent"
+	cmd.Short = "Start the addon agent"
+
+	o.AddFlags(cmd)
+	return cmd
+}
+
+// AgentOptions defines the flags for workload agent
+type AgentOptions struct {
+	HubKubeconfigFile string
+	SpokeClusterName  string
+}
+
+// NewWorkloadAgentOptions returns the flags with default value set
+func NewAgentOptions() *AgentOptions {
+	return &AgentOptions{}
+}
+
+func (o *AgentOptions) AddFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	// This command only supports reading from config
+	flags.StringVar(&o.HubKubeconfigFile, "hub-kubeconfig", o.HubKubeconfigFile, "Location of kubeconfig file to connect to hub cluster.")
+	flags.StringVar(&o.SpokeClusterName, "cluster-name", o.SpokeClusterName, "Name of spoke cluster.")
+}
+
+// RunAgent starts the controllers on agent to process work from hub.
+func (o *AgentOptions) RunAgent(ctx context.Context, controllerContext *controllercmd.ControllerContext) error {
+	hubRestConfig, err := clientcmd.BuildConfigFromFlags("" /* leave masterurl as empty */, o.HubKubeconfigFile)
+	if err != nil {
+		return err
+	}
+
+	hubKubeClient, err := kubernetes.NewForConfig(hubRestConfig)
+	if err != nil {
+		return err
+	}
+	spokeKubeClient, err := kubernetes.NewForConfig(controllerContext.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	hubKubeInformerFactory := informers.NewSharedInformerFactoryWithOptions(hubKubeClient, 10*time.Minute, informers.WithNamespace(o.SpokeClusterName))
+
+	agent := newAgentController(
+		spokeKubeClient,
+		hubKubeInformerFactory.Core().V1().ConfigMaps(),
+		o.SpokeClusterName,
+		controllerContext.EventRecorder,
+	)
+	leaseUpdater := lease.NewLeaseUpdater(
+		spokeKubeClient,
+		"helloworld",
+		"default",
+	)
+
+	go hubKubeInformerFactory.Start(ctx.Done())
+	go agent.Run(ctx, 1)
+	go leaseUpdater.Start(ctx)
+
+	<-ctx.Done()
+	return nil
+}
+
+type agentController struct {
+	spokeKubeClient    kubernetes.Interface
+	hunConfigMapLister corev1lister.ConfigMapLister
+	clusterName        string
+	recorder           events.Recorder
+}
+
+func newAgentController(
+	spokeKubeClient kubernetes.Interface,
+	configmapInformers corev1informers.ConfigMapInformer,
+	clusterName string,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &agentController{
+		spokeKubeClient:    spokeKubeClient,
+		clusterName:        clusterName,
+		hunConfigMapLister: configmapInformers.Lister(),
```

Comment:
@qiujian16 what does `hun` stand for here? Should this have been `hub` perhaps? 🤔